### DISCUSSION TITLE
includes加载子xmake.lua如何继承子xmake的一些配置？

### DISCUSSION THREAD

**Discussion by PeakRacing** [xmake-io/xmake#4156] [Q&A]
### 你在什么场景下需要该功能？

父xmake继承子xmake一些选项，例如
父xmake.lua：includes（A）
子xmake.lua：add_includedirs("test")

希望 add_includedirs("test")在父xmake里也生效

### 描述可能的解决方案

add_includedirs("test",public=true)这种？

[a comment by an automated account is not shown]

**Comment by waruqi** [comment]
目前默认的继承关系是按 tree 结构，向下继承。。子 xmake.lua 能继承 父 xmake.lua 根域的配置

另外，要看你的说的是同一个 target 内继承，还是多个 target 之间。。

target 之间继承，跟是否父子 xmake.lua 无关，仅仅跟 add_deps 和 public = true 有关。。

target 内继承追加配置，可以不同 xmake.lua 重入 target 域，追加设置就行了

[a comment by an automated account is not shown]

**Comment by PeakRacing** [comment]
> 目前默认的继承关系是按 tree 结构，向下继承。。子 xmake.lua 能继承 父 xmake.lua 根域的配置
> 
> 另外，要看你的说的是同一个 target 内继承，还是多个 target 之间。。
> 
> target 之间继承，跟是否父子 xmake.lua 无关，仅仅跟 add_deps 和 public = true 有关。。
> 
> target 内继承追加配置，可以不同 xmake.lua 重入 target 域，追加设置就行了

不是target 域，是描述域，比如子xmake.lua的描述域的有一个add_includedirs("test")我想让他一样在父xmake.lua的描述域生效

**Comment by waruqi** [comment]
我说的 target 域，就是 target 描述域

[a comment by an automated account is not shown]

**Comment by PeakRacing** [comment]
必须要add_deps么？没有用其他方法， 我现在是改第三方sdk，他是所有代码都会引用这个头，这个头不是固定位置跟随示例变动，我想在子xmake里确定它，描述域里会add_includedirs，这时候怎么让这个add_includedirs全局生效（父xmake描述域里也生效）

[a comment by an automated account is not shown]

**Comment by waruqi** [comment]
描述域里面封装 function 去 wrap add_includedirs，在父描述域的根域调用下

父

```lua
includes("common")

add_common()

target("xxx")
   -- ....
```

子

common/xmake.lua

```lua
function add_common()
    add_includedirs("xxx")
end
```
